snakemake/cli.py

```diff
@@ -1299,6 +1299,14 @@ def get_argument_parser(profiles=None):
         "network file systems. Hence, we do not spend more than a given amount of time and fall back "
         "to individual checks for the rest.",
     )
+    group_behavior.add_argument(
+        "--max-checksum-file-size",
+        type=int,
+        default=1000000,
+        metavar="BYTES",
+        help="Compute the checksum during DAG computation and job postprocessing "
+        "only for files that are smaller than the provided threshold.",
+    )
```

Comment:
Snakemake depends on the humanfriendly package, which allows to parse sizes like this from strings. I think we should use that here. You can pass humanfriendly.parse_size to the argument so that it gets directly interpreted and translated into bytes when the user specifies somehing like `--max-checksum-file-size 1MB`.